### ISSUE TITLE
Allow to add output directory/path to fit()

### ISSUE THREAD

**Issue by marianklose** [pharmpy/pharmpy#2257]
It would be great if we could specify where the output of a fit() run (modelfit_dir) is being saved. Currently if I run

```
# load package
library(pharmr)

# read in model
model <- read_model('./PATH/TO/LOCATION/run001.mod')

# estimate
est <- model |> fit()
```

then the output folder modelfit_dir from NONMEM is being saved at my current working directory (`getwd`) instead of './PATH/TO/LOCATION/'.  I would've expected that the folder is being created at the location of the model file. A workaround is to change the working directory prior to `fit()`, but messing with the wd is not always the best way to go.  

Is this the expected behaviour? And could you consider adding an argument to fit() allowing to specify the output dir? FYI: I am currently on version `0.103.0`

Thanks a lot for your great work!

**Comment by marianklose** [comment]
Amazing, thanks for the quick response. Keep up the good work!

**Comment by stellabelin** [comment]
No problem, thanks for the input! And sorry, wrote the workaround wrong! It should be `run_modelfit(model, path='path/to/dir')`

**Comment by marianklose** [comment]
@stellabelin I guess I need to come back to this issue again, maybe we can reopen it. The added path functionality generally works fine, thanks for that! However to effectively use it I would need to create a function to dynamically create the name for the output folder. 

If I use the `fit()` function without any path argument I would get with every run a folder like this:

- wd/modelfit_dir1
- wd/modelfit_dir2
- wd/modelfit_dir3

with wd as the working dir. If I use the path argument like this:

```
# perform NONMEM estimation
est_out <- pharmr::fit(mod, path = "./output")
```

it works for the first run, but then I get an error if I run it again:

```
FileExistsError: [WinError 183] Cannot create a file when that file already exists: 'output'
```

So I would need to write a custom functions to add a suffix to the folder depening on the existent number of folders. Maybe I can shortly explain where I am coming from. Typically I organize my folders into something like this:

- ./scripts
- ./models
- ./output

Some consider it good practice to keep scripts and output seperated. The whole idea would be to have the working directory somewhere in your project, then to use an R (or python) script inside `./scripts` to read in a model from `./models` (which already works) and to save all the modeling output in `./output`. The `./output` directory should then be populated with the `modelfit_dir{suffix}` folders. 

So in a nutshell my request would be to be able to specify the general output directory where the `modelfit_dir{suffix}` folders are being stored instead of explicitely naming them. Do you think this would be possible? Thanks for considering it!

**Comment by stellabelin** [comment]
Hi! Thank you for the input, I agree that this sort of functionality could be useful! We are in the process of redesigning how the tool directories work in general, we can think about how this use case can fit into the new system!
